src/wallet/nep2.js

```diff
@@ -71,4 +105,38 @@ export const decrypt = (encryptedKey, keyphrase, scryptParams = DEFAULT_SCRYPT)
   return account.WIF
 }
 
+/**
+ * Decrypts an encrypted key using a given keyphrase under NEP-2 Standard.
+ * @param {string} encryptedKey - The encrypted key (58 chars long).
+ * @param {string} keyphrase - The password will be encoded as UTF-8 and normalized using Unicode Normalization Form C (NFC).
+ * @param {scryptParams} [scryptParams] - Parameters for Scrypt. Defaults to NEP2 specified parameters.
+ * @returns {string} The decrypted WIF key.
+ */
+export const decryptAsync = (encryptedKey, keyphrase, scryptParams = DEFAULT_SCRYPT_ASYNC) => {
+  return new Promise((resolve, reject) => {
+    scryptParams = ensureScryptParams(scryptParams)
```

Comment:
`ensureScryptParams` ensures the fields cost, parallel and blockSize are filled. This does nothing for you because you are using n,r,p.